samples/nrf9160/nrf_cloud_rest_cell_location/src/main.c

```diff
@@ -195,15 +211,49 @@ static void get_cell_info(void)
 	} else {
 		struct lte_lc_ncellmeas_params ncellmeas_params = {
 			.search_type = search_type,
+			.gci_count = CONFIG_REST_CELL_GCI_COUNT
 		};
 
-		LOG_INF("Requesting neighbor cell measurement...");
+		/* Set the result buffers */
 		cell_info.neighbor_cells = neighbor_cells;
+		cell_info.gci_cells = gci_cells;
+
+		LOG_INF("Requesting neighbor cell measurement");
 		err = lte_lc_neighbor_cell_measurement(&ncellmeas_params);
 		if (err) {
 			LOG_ERR("Failed to start neighbor cell measurement, error: %d", err);
+		} else {
+			LOG_INF("Waiting for measurement results...");
+		}
+	}
+}
+
+static void process_cell_pos_type_change(void)
+{
+	if (!ready) {
+		return;
+	}
+
+	/* Toggle active cell pos type */
```

Comment:
Just general comment because didn't fully understand logics here.
What are we meaning by LOCATION_TYPE_MULTI_CELL? 
Anyways, with GCI search types,  in my testings, neighbors for the serving cell are returned only when RRC active, and GCI cells when RRC idle.
With legacy search types, neighbors for the serving cell could be returned in either state.

Reply:
this sample demonstrates single cell and multi-cell requests.  it uses the LOCATION_TYPE enum to track which type is active.
this function just switches to the other mode and initiates a reading/request.

the RRC mode handling was a result of [this discussion](https://github.com/nrfconnect/sdk-nrf/pull/8775#discussion_r975867618).
i rarely, if ever, get neighbor cell results.  i am not sure why.  i do get GCI results though.

the AT cmd manual states the following:
```
Neighboring cell measurements are available during a data or signaling RRC connection only when the
network has configured measurement rules for the device and the configured criteria is met.
If neighboring cell measurements have been active before the connection, measurement results from an
RRC idle mode are available in connected mode for 10 seconds after the last measurement in idle mode
has passed.
Measurements in idle mode are based on generic configurations from the network and radio conditions.
Usually, measurements are not performed in idle mode during good radio conditions unless they are
requested with the %NCELLMEAS command.
If the network configures connected mode measurements, measurement results from idle mode are
cleared before the 10 seconds has passed.
```
so it sounds like by requesting an NCELLMEAS during idle mode should be ok.